File: klinara-android/app/src/main/java/com/klinara/android/designsystem/components/KlinaraStatCard.kt
package com.klinara.android.designsystem.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.IntrinsicSize
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxHeight
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.DateRange
import androidx.compose.material.icons.filled.ShoppingCart
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.Immutable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.semantics.clearAndSetSemantics
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import com.klinara.android.designsystem.KlinaraMetrics
import com.klinara.android.designsystem.KlinaraPreviews
import com.klinara.android.designsystem.KlinaraTheme
import com.klinara.android.designsystem.KlinaraType

/** Tek sayılık özet kartının verisi. `value == null` → "—" (izin var, sayı yok). */
@Immutable
data class KlinaraStat(
    val label: String,
    val value: String?,
    val icon: ImageVector? = null,
    val hint: String? = null,
)

/**
 * Tek sayılık özet kartı — web `StatCard` karşılığı: üstte etiket + ikon, ortada büyük değer,
 * altta ipucu (geçen aya göre değişim gibi).
 *
 * **Yükseklik içerikten BAĞIMSIZ.** Şeritteki kartlar yan yana ve alt alta hizalı durmalı; ipucu
 * olan kartla olmayan, ipucu bir satıra sığanla sığmayan farklı boyda çıkıyordu. Satır bütçesi
 * sabit: etiket 1, değer 1, ipucu HER ZAMAN 2 satır yer tutar (boşsa da). Sabit dp yükseklik
 * yerine satır bütçesi: büyük yazı boyunda (fontScale 2.0) kart kırpılmadan birlikte büyür.
 *
 * TalkBack kartı TEK öğe olarak okur ("Bu ay ciro, 12.500,00 ₺, Geçen aya göre +%8"): üç ayrı
 * odak durağı bir sayıyı üç parçaya bölerdi.
 */
@Composable
fun KlinaraStatCard(
    stat: KlinaraStat,
    modifier: Modifier = Modifier,
    isLoading: Boolean = false,
) {
    val colors = KlinaraTheme.colors
    val shape = RoundedCornerShape(KlinaraMetrics.cardRadius)
    val value = stat.value ?: "—"

    Column(
        modifier =
            modifier
                .clip(shape)
                .background(colors.surfaceRaised)
                .border(KlinaraMetrics.borderWidth, colors.border, shape)
                .padding(KlinaraMetrics.md)
                .clearAndSetSemantics {
                    contentDescription = listOfNotNull(stat.label, if (isLoading) "Yükleniyor" else value, stat.hint)
                        .joinToString(", ")
                },
        verticalArrangement = Arrangement.spacedBy(KlinaraMetrics.sm),
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Text(
                KlinaraType.labelText(stat.label),
                style = KlinaraType.label,
                color = colors.charcoalMuted,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
                modifier = Modifier.weight(1f),
            )
            stat.icon?.let {
                Icon(it, contentDescription = null, tint = colors.sageDeep, modifier = Modifier.size(ICON_SIZE))
            }
        }
        // Yer tutucu değerin KENDİ satır yüksekliğini kullanır: yükleme bitince kart boyu değişmez.
        Box(if (isLoading) Modifier.width(PLACEHOLDER_WIDTH) else Modifier) {
            Text(
                value,
                style = KlinaraType.titleM,
                color = if (isLoading) Color.Transparent else colors.charcoal,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
            )
            if (isLoading) {
                Box(
                    Modifier
                        .matchParentSize()
                        .clip(RoundedCornerShape(KlinaraMetrics.xs))
                        .background(colors.border.copy(alpha = 0.5f)),
                )
            }
        }
        Text(
            stat.hint.orEmpty(),
            style = KlinaraType.bodyM,
            color = colors.charcoalMuted,
            minLines = HINT_LINES,
            maxLines = HINT_LINES,
            overflow = TextOverflow.Ellipsis,
        )
    }
}

/**
 * Özet kartı şeridi — iki sütunlu ızgara. Telefon genişliğinde dört kart yan yana sığmaz; web de
 * dar ekranda iki sütuna iner. Tek kalan kart satırın yarısında durur, genişlemez: kart boyları
 * birbirinden zıplamasın.
 */
@Composable
fun KlinaraStatStrip(
    stats: List<KlinaraStat>,
    modifier: Modifier = Modifier,
    isLoading: Boolean = false,
) {
    Column(modifier = modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(KlinaraMetrics.sm)) {
        stats.chunked(COLUMNS).forEach { row ->
            Row(
                // Satır bütçesi zaten eşit boy veriyor; IntrinsicSize yalnız bir etiketin cihaz
                // fontunda beklenmedik şekilde yükseldiği durumda komşuyu da hizalı tutar.
                modifier = Modifier.fillMaxWidth().height(IntrinsicSize.Min),
                horizontalArrangement = Arrangement.spacedBy(KlinaraMetrics.sm),
            ) {
                row.forEach { stat ->
                    KlinaraStatCard(
                        stat = stat,
                        isLoading = isLoading,
                        modifier = Modifier.weight(1f).fillMaxHeight(),
                    )
                }
                repeat(COLUMNS - row.size) { Column(Modifier.weight(1f)) {} }
            }
        }
    }
}

private const val COLUMNS = 2
private val ICON_SIZE = 16.dp
private val PLACEHOLDER_WIDTH = 96.dp
private const val HINT_LINES = 2

@KlinaraPreviews
@Composable
private fun KlinaraStatStripPreview() {
    KlinaraTheme {
        KlinaraStatStrip(
            stats =
                listOf(
                    KlinaraStat("Bugünkü randevu", "12", Icons.Filled.DateRange, "8 aktif · 4 tamamlandı"),
                    KlinaraStat("Bu ay ciro", "182.500,00 ₺", Icons.Filled.ShoppingCart, "Geçen aya göre +%8"),
                    KlinaraStat("Bu ay doluluk", null),
                ),
        )
    }
}

File: klinara-ios/klinara-ios/DesignSystem/Components/KlinaraStatCard.swift
import SwiftUI

/// Tek sayılık özet kartının verisi. `value == nil` → "—" (izin var, sayı yok).
struct KlinaraStat: Identifiable {
    let label: String
    let value: String?
    var icon: String?
    var hint: String?

    var id: String { label }
}

/// Tek sayılık özet kartı — web `StatCard` karşılığı: üstte etiket + ikon,
/// ortada büyük değer, altta ipucu (geçen aya göre değişim gibi).
///
/// **Yükseklik içerikten BAĞIMSIZ.** Şeritteki kartlar hizalı durmalı; ipucu
/// olan kartla olmayan, ipucu bir satıra sığanla sığmayan farklı boyda
/// çıkıyordu. Satır bütçesi sabit: etiket 1, değer 1, ipucu HER ZAMAN 2 satır
/// yer tutar (`reservesSpace`). Sabit pt yükseklik yerine satır bütçesi: büyük
/// Dynamic Type boyunda kart kırpılmadan birlikte büyür.
///
/// VoiceOver kartı TEK öğe olarak okur: üç ayrı odak durağı bir sayıyı üç
/// parçaya bölerdi.
struct KlinaraStatCard: View {

    let stat: KlinaraStat
    var isLoading = false

    /// İpucu alanı: iki `footnote` satırı. `reservesSpace` tek başına yetmedi —
    /// markanın satır aralığı yalnız GERÇEKTEN sarılan metinde ekleniyor ve iki
    /// satırlık ipucu kartı komşusundan 3pt uzun yapıyordu. Yükseklik Dynamic
    /// Type ile birlikte ölçekleniyor.
    @ScaledMetric(relativeTo: .footnote) private var hintHeight: CGFloat = 48

    var body: some View {
        VStack(alignment: .leading, spacing: KlinaraMetrics.sm) {
            // İkon satır yüksekliğine KATILMIYOR (overlay): SF Symbol'lerin
            // yüksekliği sembolden sembole değişiyor ve rozetli bir ikon kartı
            // komşusundan uzun yapıyordu.
            Text(stat.label.uppercased(with: Locale(identifier: "tr_TR")))
                .klinaraText(.label)
                .foregroundStyle(KlinaraColor.charcoalMuted)
                .lineLimit(1)
                .minimumScaleFactor(0.85)
                .padding(.trailing, stat.icon == nil ? 0 : 22)
                .frame(maxWidth: .infinity, alignment: .leading)
                .overlay(alignment: .trailing) {
                    if let icon = stat.icon {
                        Image(systemName: icon)
                            .font(.system(size: 13, weight: .semibold))
                            .foregroundStyle(KlinaraColor.sageDeep)
                            .frame(width: 18)
                    }
                }

            // Yer tutucu değerin KENDİ satır yüksekliğini kullanır: yükleme
            // bitince kart boyu değişmez.
            Text(stat.value ?? "—")
                .klinaraText(.titleM)
                .monospacedDigit()
                .foregroundStyle(KlinaraColor.charcoal)
                .lineLimit(1)
                .minimumScaleFactor(0.7)
                .opacity(isLoading ? 0 : 1)
                .overlay(alignment: .leading) {
                    if isLoading {
                        RoundedRectangle(cornerRadius: KlinaraMetrics.xs)
                            .fill(KlinaraColor.border.opacity(0.5))
                            .frame(width: 96)
                    }
                }

            Text(stat.hint ?? "")
                .klinaraText(.bodyM)
                .font(.footnote)
                .lineSpacing(0)
                .foregroundStyle(KlinaraColor.charcoalMuted)
                .lineLimit(2)
                .frame(maxWidth: .infinity, minHeight: hintHeight, maxHeight: hintHeight, alignment: .topLeading)
        }
        .padding(KlinaraMetrics.md)
        // Izgara satırının yüksekliğini doldur: bir etiket beklenmedik şekilde
        // yükselirse komşu kart da hizalı kalır.
        .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .topLeading)
        .background(KlinaraColor.surfaceRaised)
        .overlay(
            RoundedRectangle(cornerRadius: KlinaraMetrics.cardRadius)
                .stroke(KlinaraColor.border, lineWidth: KlinaraMetrics.borderWidth)
        )
        .clipShape(.rect(cornerRadius: KlinaraMetrics.cardRadius))
        .accessibilityElement(children: .ignore)
        .accessibilityLabel(
            [stat.label, isLoading ? "Yükleniyor" : (stat.value ?? "—"), stat.hint]
                .compactMap { $0 }
                .joined(separator: ", ")
        )
    }
}

/// Özet kartı şeridi — iki sütunlu ızgara. Telefon genişliğinde dört kart yan
/// yana sığmaz; web de dar ekranda iki sütuna iner. Kartlar satır bütçesiyle
/// eşit boyda; tek kalan kart satırın yarısında durur.
struct KlinaraStatStrip: View {

    let stats: [KlinaraStat]
    var isLoading = false

    var body: some View {
        Grid(alignment: .topLeading, horizontalSpacing: KlinaraMetrics.sm, verticalSpacing: KlinaraMetrics.sm) {
            ForEach(Array(stride(from: 0, to: stats.count, by: 2)), id: \.self) { start in
                GridRow {
                    KlinaraStatCard(stat: stats[start], isLoading: isLoading)
                    if start + 1 < stats.count {
                        KlinaraStatCard(stat: stats[start + 1], isLoading: isLoading)
                    } else {
                        Color.clear
                    }
                }
            }
        }
    }
}

#Preview("Özet şeridi") {
    KlinaraStatStrip(stats: [
        KlinaraStat(label: "Bugünkü randevu", value: "12", icon: "calendar", hint: "8 aktif · 4 tamamlandı"),
        KlinaraStat(label: "Bu ay ciro", value: "182.500,00 ₺", icon: "banknote", hint: "Geçen aya göre +8%"),
        KlinaraStat(label: "Bu ay doluluk", value: nil, icon: "gauge.with.needle"),
    ])
    .padding()
    .background(KlinaraColor.surface)
}
